### ISSUE TITLE
Kubernetes kubectl exec timeout

### ISSUE THREAD

**Issue by bluemalkin** [rancher/rancher#8579]
**Rancher Versions:**
Server: v1.5.5
kubernetes: 1.5.4-rancher1-1

**Environment Type:**
Kubernetes on AWS. Tried with both overlapping planes and resiliency planes.

**Steps to Reproduce:**
Run `kubectl exec -it POD -- COMMAND` either in Rancher CLI or locally using the imported `.kubectl/config`

**Results:** 
The command doesn't respond and gets a timeout.

The only working way for me to shell into a container is by going to Infrastructure, Containers, Execute shell on a container.

**Comment by aemneina** [comment]
@tombmurphy what kind of firewall/lb do you have in front of rancher?

**Comment by tombmurphy** [comment]
@aemneina it's deployed on AWS so it uses Security Groups. There's no lb.

Do kubectl exec commands go via a different port ?

**Comment by galal-hussein** [comment]
@tombmurphy Kubectl exec sends POST requests to the Kubernetes API, for example to exec ls command:
```
POST  https://x.x.x.x:8080/r/projects/1a36/kubernetes/api/v1/namespaces/default/pods/mariadb-mysql-817230144-5k2s7/exec?command=ls&container=mariadb-mysql&container=mariadb-mysql&stdin=true&stdout=true&tty=true
```

Can you increase the log level when you run the command and paste the logs:

`kubectl --v=7 exec -it POD -- COMMAND`

**Comment by superseb** [comment]
Please report back if we can help.

**Comment by fabiojvalente** [comment]
Hello,

Just faced this issue on v1.6.14 

The entire stack is running on AWS, after enabling the logging as suggested by @galal-hussein which returned this message:

``
F0213 15:36:33.625761   82541 helpers.go:120] Error from server: error dialing backend: dial tcp [rancher_node_running_pod_ip]:10250: getsockopt: connection timed out
``

Opening port 10250 on the `rancher-maghine` SG created by rancher when adding new hosts fixed it

**Comment by martindanko** [comment]
@fabiojvalente - just interested when you say opened 10250, do you mean open it to 0.0.0.0/0 ?

**Comment by fabiojvalente** [comment]
it was on aws, and I had to open the port to hosts on the same security group

The laziest fix would be opening it to 0.0.0.0/0 Yes

**Comment by martindanko** [comment]
I'm experimenting and if I do the same, open it to hosts on the same security group and my remote IP, I get timeouts, when temporally testing with opening on *0.0.0.0/0*  I don't. Would you know why? Thanks for answering.
